### DISCUSSION TITLE
QuestionnaireForm "response" prop so it's possible to edit an existing response

### DISCUSSION THREAD

**Discussion by filipeximenes** [medplum/medplum#5586] [Ideas]
Hi folks, I need to build a form a Questionnaire that is able to edit an existing QuestionnaireResponse. The QuestionnaireForm component seems to have most of what I need but it doesn't accept a QuestionnaireResponse as a prop. I did some very basic tests and It seems like just by initializing the response variable with a passed prop would already make it work
```
const [response, setResponse] = useState<QuestionnaireResponse | undefined>(props.response);
```
I assume there's some reason for this not be implemented yet, perhaps there's some internal component complexity to this that I'm not aware of. Please LMK if this makes sense to you and what would the best way to add this behavior.

**Comment by codyebberson** [comment]
Hi @filipeximenes  - you're absolutely correct, this is an important feature request, and definitely something that we intend to add.  We recently hired additional staff to work on front end components, and `<QuestionnaireForm>` is coming up soon.

**Reply by filipeximenes** [reply]
Excellent, thanks for getting back about this
